webapp/app/routes/app/emergency/new/_route.tsx

```diff
@@ -0,0 +1,139 @@
+import { emergencyBoardAddress } from "@/.server/service/authorized-users";
+import {
+  saveEmergencyProposal,
+  validateEmergencyProposalCalls,
+} from "@/.server/service/emergency-proposals";
+import { hexSchema } from "@/common/basic-schemas";
+import { type Call, callSchema } from "@/common/calls";
+import { StepsWizard, WizardStep } from "@/components/steps-wizard";
+import { NewEmergencyProposalStep1, type Step1 } from "@/routes/app/emergency/new/step1";
+import { NewEmergencyProposalStep2 } from "@/routes/app/emergency/new/step2";
+import { Step3 } from "@/routes/app/emergency/new/step3";
+import { requireUserFromHeader } from "@/utils/auth-headers";
+import { badRequest } from "@/utils/http";
+import { type ActionFunctionArgs, json } from "@remix-run/node";
+import { useFetcher, useLoaderData, useNavigate } from "@remix-run/react";
+import { useEffect, useState } from "react";
+import { getFormData } from "remix-params-helper";
+import { $path } from "remix-routes";
+import { z } from "zod";
+
+export async function loader() {
+  return json({ emergencyBoardAddress: await emergencyBoardAddress() });
+}
+
+export async function action({ request }: ActionFunctionArgs) {
+  const user = requireUserFromHeader(request);
+  const body = await getFormData(
+    request,
+    z.object({
+      intent: z.enum(["validate", "save"]),
+      calls: z.string(),
+      salt: hexSchema,
+      title: z.string().min(1),
+    })
+  );
+
+  if (!body.success) {
+    throw badRequest(body.errors.toString());
+  }
+
+  const calls = z.array(callSchema).safeParse(JSON.parse(body.data.calls));
```

Comment:
It seems here these 2 parsers can be put together. If we lose the JSON.stringify in the call as proposed this could be greatly simplified.